### PR TITLE
chore(deps) fix V8 release builds

### LINKED ISSUE #485: Fix V8 builds releases
V8 builds are currently broken in:
- Ubuntu 18.04
- CentOS 8
- ArchLinux

See: https://github.com/Kong/ngx_wasm_module/actions/runs/7691838744

### PR DESCRIPTION
**Pull request by hishamhm**
Closes #485.

- [x] fix Ubuntu 18.04
- [x] fix CentOS 8
- [x] fix ArchLinux

### CODE REVIEW

**Comment by thibaultcha** [comment]
It seems there still are a couple of problems as per the test run in #488?

**Comment by hishamhm** [comment]
@thibaultcha ah, yes, I saw the green checkmark and undrafted this too soon. The [actual test](https://github.com/Kong/ngx_wasm_module/actions/runs/7702502086) I triggered manually yesterday before calling it a night gave the same results as #488.

File: config [hunk 1]
```diff
@@ -31,7 +31,11 @@ case $ngx_wasm_runtime_name in
         ngx_wasm_runtime_srcs="$ngx_addon_dir/src/wasm/wrt/ngx_wrt_v8.c"
         ngx_wasm_runtime_lib_name="wee8"
         ngx_wasm_runtime_pre_libs="-lv8bridge"
-        ngx_wasm_runtime_post_libs="-ldl -lm -lpthread -lstdc++ -latomic"
+        ngx_wasm_runtime_post_libs="-ldl -lm -lpthread -lstdc++"
+
+        if [ "$NGX_WASM_RUNTIME_V8_NEEDS_LIBATOMIC" = 1 ]; then
```
Comment (thibaultcha) [inline config:36]: Could we automatize this with `$NGX_PLATFORM` instead? Nginx's `auto/` scripts already rely on it: https://github.com/nginx/nginx/blob/master/auto/os/conf#L22

**Comment by hishamhm** [comment]
This is ready for review/merge.

* Turns out that the libatomic thing was OS-specific after all! 
* I also needed more dancing around to update GCC for CentOS 8
* older distros (CentOS 7 and Ubuntu 18.04) need actions/checkout downgraded to v3 due to a binary incompatibility with Node 20.
* there's some overlap with @casimiro's work on #506 because I needed upload-artifacts fixed in order to get a full run.
* Here's a manually-triggered `release.yml` run showing all builds green: https://github.com/Kong/ngx_wasm_module/actions/runs/7908368436

(Note that if two different branches working on `release.yml` trigger at the same time, you might get the wrong build container and cause a build failure — this happened to me yesterday when both @casimiro and I were doing `release.yml` tests)

**Comment by thibaultcha** [comment]
Merged in #510.

**Pull request closed without merging**